### PR TITLE
Add a fixit for attempting member lookup on Any.

### PR DESCRIPTION
**Pull request by rudkx**
#### What's in this pull request?

Add a fixit for attempting member lookup on Any.
#### Resolved bug number: ([rdar://problem/27639935](rdar://problem/27639935))

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| All supported platforms | @swift-ci Please smoke test and merge |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

A smoke test on macOS does the following:
1. Builds the compiler incrementally.
2. Builds the standard library only for macOS. Simulator standard libraries and
   device standard libraries are not built.
3. lldb is not built.
4. The test and validation-test targets are run only for macOS. The optimized
   version of these tests are not run.

A smoke test on Linux does the following:
1. Builds the compiler incrementally.
2. Builds the standard library incrementally.
3. lldb is built incrementally.
4. The swift test and validation-test targets are run. The optimized version of these
   tests are not run.
5. lldb is tested.

**Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| All supported platforms | @swift-ci Please test and merge |
| OS X platform | @swift-ci Please test OS X platform |
| OS X platform | @swift-ci Please benchmark |
| Linux platform | @swift-ci Please test Linux platform |

**Lint Testing**

| Language | Comment |
| --- | --- |
| Python | @swift-ci Please Python lint |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

Suggest casting to AnyObject or a more specific type.

rdar://problem/27639935
(cherry picked from commit 56a5ef00a4234966698dda1d0aedded170a0b305)

### CODE REVIEW

**Comment by rudkx** [comment]
@swift-ci Smoke test.

**Comment by rudkx** [comment]
@jckarter Can you review for the swift-3.0-branch?

**Comment by jckarter** [comment]
Looks great @rudkx, thanks for fixing this!

File: lib/Sema/CSDiag.cpp [hunk 1]
```diff
@@ -2477,6 +2477,14 @@ diagnoseUnviableLookupResults(MemberLookupResult &result, Type baseObjTy,
           diagnose(loc, diag::did_you_mean_raw_type);
           return; // Always prefer this over typo corrections.
         }
+      } else if (baseObjTy->isAny()) {
+        if (auto DRE = dyn_cast<DeclRefExpr>(baseExpr)) {
+          auto name = DRE->getDecl()->getName().str().str();
+          std::string replacement = "(" + name + " as AnyObject)";
```
Comment (jrose-apple) [inline lib/Sema/CSDiag.cpp]: Nitpick: an `llvm::SmallString` would probably be better here.

Comment (jrose-apple) [inline lib/Sema/CSDiag.cpp]: Also, why limit the note to DeclRefExprs? I imagine one of the most common cases of this is "drilling into a plist/JSON dictionary or array".

Comment (rudkx) [inline lib/Sema/CSDiag.cpp]: @jrose-apple I'll take a look at doing something more general.

Comment (rudkx) [inline lib/Sema/CSDiag.cpp]: @jrose-apple I've pushed a PR for master. Please take a look and I'll put up a PR for swift-3.0-branch if it tests and merges successfully.

https://github.com/apple/swift/pull/4116

**Comment by rudkx** [comment]
I'll push a more general fix for swift-3.0-branch.

**Pull request closed without merging**